pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractDispatcherMultipleConsumers.java

```diff
@@ -119,26 +120,30 @@ public Consumer getNextConsumer() {
             return null;
         }
 
-        if (currentConsumerRoundRobinIndex >= consumerList.size()) {
-            currentConsumerRoundRobinIndex = 0;
+        // other thread will change consumerList
+        if (currentConsumerRoundRobinIndex.get() >= consumerList.size()) {
+            currentConsumerRoundRobinIndex.set(0);
         }
 
-        int currentRoundRobinConsumerPriority = consumerList.get(currentConsumerRoundRobinIndex).getPriorityLevel();
+        int currentRoundRobinConsumerPriority = consumerList.get(currentConsumerRoundRobinIndex.get())
```

Comment:
Still, there is race condition between Line 124 and Line 128, if `consumerList.size` decreases.

Reply:
do you have any advice for this ?

Reply:
Have a peek of the code. It seems all modification of consumerList is in synchronized method(You need go through all the usage to verify this). 
If so, this index can be calculated in a synchronized block.